### ISSUE TITLE
Folder only syncs if web UI is opened

### ISSUE THREAD

**Issue by wiwie** [syncthing/syncthing-android#1172]
If your issue is a bug report, replace this boilerplate with a description of the problem, being sure to include at least:

## what happened
folder only syncs with other devices if I open the web UI in the syncthing android client.

##    what you expected to happen instead
the sync should start immediately after adding the sync, without having to open the web UI

##    any steps to reproduce the problem.
 -> Opened syncthing android client
 -> Added a new local folder as sync
 -> Added some remote devices for sync
 -> waiting for folder to be synced for other devices
 -> other devices get "new folder" request
 -> confirm request
 -> no sync is happening
 -> opening web UI on syncthing android
 -> sync starts instantly

### Version Information

    App Version: 0.10.11
    Syncthing Version: v0.14.48
    Android Version: Android 8.1.0

**Comment by AudriusButkevicius** [comment]
Syncing is not immediate, they have to connect, exchange indexes etc, so I think it's just a coincidence that something happened as you opened the web ui.

**Comment by wiwie** [comment]
I observed this behaviour several times already. Opening the web UI immediately starts the sync process.

This time I also waited a subjectively sufficient amount of time (>10mins) before trying the web UI.

This workaround works a little too reliable to be coincidence IMO.

Can this behaviour be explained somehow? Does opening the web UI trigger an immediate reconnect to other clients for instance?

**Comment by AudriusButkevicius** [comment]
No, web ui triggers nothing. The web ui is running already even if not opened. I suggest you look into the logs but I am fairly positive it's just a coincidence. Perhaps you are mixing the general ui with web ui, and if the app is not open in general it gets killee by android.

**Comment by Catfriend1** [comment]
Yes, some logs would be very nice with mentioning when the sync happened (datetimestamp mention) from your point of view. For me sync starts nearly immediately after the folder has been added on both sides. If I cant wait this startoff minute as devices may first have to reconnect which takey some time affer the accept I'll restart one side (android). Another thesis would be problems with the local discoveries (if you havent configured static ip addresses) being partly blocked on your Android version as google does a lot against permanently broadcasting apps.

**Comment by Catfriend1** [comment]
Btw technical:
 I noticed we are starting the syncthing service with WebGuiActivity.
This does only occur if the app has been relaunched from android recent apps after pressing home turning back or after force closing the app. The start will only be executed down to the service if the service is in stopped state when the activity is created (service is a Singleton). This should happen rarely because most users were in Mainactivity before they go to webactivity. It must happen expectedly when a user has ticked the 'directly go into web ui on app start' option. 

> startService(new Intent(this, SyncthingService.class));

**Comment by Catfriend1** [comment]
I suspect syncthing to fastly crash out after the native binary came up like I mentioned in issue #1174. But we definitely need logs for proof.

**Comment by Catfriend1** [comment]
Related: https://forum.syncthing.net/t/android-0-10-11-not-syncing-device-id-gone/11781/17
Put on my todo list to better show syncthing crashes or terminations caused by android to the user. Maybe also a restart mechanism...

**Comment by Catfriend1** [comment]
This will be partly fixed by the new status tab that can later show if the syncthing native crashed. Refer to: https://github.com/syncthing/syncthing-android/pull/1205
